### ISSUE TITLE
Using the same information in two places in the template is fails with 0.3.0

### ISSUE THREAD

**Issue by cpaulik** [pytroll/trollsift#11]
Trying to e.g. use a version identifier/date etc. from the filename also in a path fails with version 0.3.0

The following code illustrates the problem:

```python
import trollsift

test_template = '{version_number:1s}/filename_with_version_number_{version_number:1s}.tif'
p = trollsift.Parser(test_template)
print(p.parse('1/filename_with_version_number_1.tif'))
```

The output of the different versions:

## version 0.2.0

```python
{'version_number': '1'}
```

## version 0.3.0
```python
Traceback (most recent call last):
  File "01-175034.py", line 5, in <module>
    print(p.parse('1/filename_with_version_number_1.tif'))
  File "/home/cpa/.pyenv/versions/miniconda2-4.3.30/envs/libsat/lib/python2.7/site-packages/trollsift/parser.py", line 49, in parse
    return parse(self.fmt, stri)
  File "/home/cpa/.pyenv/versions/miniconda2-4.3.30/envs/libsat/lib/python2.7/site-packages/trollsift/parser.py", line 346, in parse
    keyvals = regex_formatter.extract_values(fmt, stri)
  File "/home/cpa/.pyenv/versions/miniconda2-4.3.30/envs/libsat/lib/python2.7/site-packages/trollsift/parser.py", line 278, in extract_values
    match = re.match(regex, stri)
  File "/home/cpa/.pyenv/versions/miniconda2-4.3.30/envs/libsat/lib/python2.7/re.py", line 141, in match
    return _compile(pattern, flags).match(string)
  File "/home/cpa/.pyenv/versions/miniconda2-4.3.30/envs/libsat/lib/python2.7/re.py", line 251, in _compile
    raise error, v # invalid expression
sre_constants.error: redefinition of group name 'version_number' as group 2; was group 1
```

**Comment by djhoese** [comment]
I know @pnuu had had a similar issue, but it turned out his pattern was incorrect so we ignored it. My main question for you @cpaulik is, if this is a desired usage then what gets returned if the values are different or formatted differently? The parsing code is now using regular expressions so having it match a previous group/field shouldn't be a huge issue, but I'm wondering if there are some edge cases. Do you expect the values returned/found by parsing to be *exactly* the same?

**Comment by djhoese** [comment]
Also the obvious workaround until I fix this is to use a different name for each repeat of the parameter.

**Comment by cpaulik** [comment]
My assumption was that when the version numbers are different a error would be raised since the name does not conform to the template. But it seems this assumption was wrong in the first place. 
 
Another thing I would like in general if compose/parse would be symmetrical. So since compose works with this template then I would also like to be able to parse it.

**Comment by djhoese** [comment]
I agree on things being symmetrical and I'm not disagreeing that this should be supported. I'm not sure *everything* can be symmetrical without needing a ridiculous amount of work (formatting the parameter a certain way and then un-formatting it), but I can make this case work.

**Comment by djhoese** [comment]
@cpaulik See referenced PR above (#12).

**Comment by cpaulik** [comment]
Thank you for the quick fix.

**Comment by djhoese** [comment]
This is now available in 0.3.1 which is on PyPI. The conda-forge package will be updated this weekend (waiting on their bot to automatically pick up on the new release, make a pull request, test the build, merge the PR, then do the actual build).

**Comment by cpaulik** [comment]
Thank you. That's amazing
